test/logging/logging_test.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logging
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+)
+
+type abc struct {
+	A int
+	b string
+	C *de
+	F func()
+}
+
+type de struct {
+	D string
+	e float64
+}
+
+const runFailingTests = false
```

Comment:
Why is this const?

Reply:
The problem with testing the methods which run and cause test failures, is they cause the unit tests to fail (since testing.T doesn't have "expected failures"). We can't create struct testing.T objects, so we can't mock it. The only way I can otherwise think to test this is put it into a helper function, then have the test call `go test` with a separate test flag to enable failing tests, then parse the output of `go test`. Given the simplicity of the calls in comparison to their components which are unit tested in other ways, I did not think the last method was worth the effort. However, I did want to leave the ability for developers to run the tests manually and observe the results.
What do you think? If you think this is the right choice, I can add it as a comment (I should have anyway).

Reply:
Victor suggested adding debug constructor which doesn't call testing.T's fail. Will do.

Reply:
Done.